extensions/yafw/src/index.tsx

```diff
@@ -0,0 +1,144 @@
+    if (isFileBeingRemoved) {
+      const fileBeingRemoved = files.find((file) => !newFiles.includes(file));
+      setRemovedFiles([...removedFiles, fileBeingRemoved!]);
+    } else if (automaticallyAdded) {
+      setAutoLoaded(true);
+      // avoid adding files that we automatically added but the user removed
+      newFiles = newFiles.filter((file) => !removedFiles.includes(file));
+    } else {
+      // all new legit files should be deleted from the removed files list
+      setRemovedFiles(removedFiles.filter((file) => !newFiles.includes(file)));
+    }
+
+    newFiles = newFiles.filter(fileExists);
+
+    return _setFiles(newFiles);
+  };
+
+  useEffect(() => {
+    const lookForVideoSources = () => {
+      Clipboard.read().then(setClipboard);
+      getSelectedFinderItems()
+        .then(setSelectedFinderItems)
+        .catch(() => setSelectedFinderItems([]));
+    };
+    lookForVideoSources();
+    /* Poll for new files every 3 seconds
+     * In case the view is left in the background (i.e Raycast loses focus)
+     * the following warning will be shown at some point:
+     * > Looks like the command is rendering a lot without any changes. This might indicate an rendering loop caused by a `setState` not being wrapped in a `useEffect`.
+     * > This will degrade the performances of Raycast and Raycast might arbitrarily decide to terminate the extension if it happens too many times.
+     */
+    setInterval(lookForVideoSources, 3000);
+  }, []);
+
+  useEffect(() => {
+    if (clipboard && "file" in clipboard && !files.includes(clipboard.file!)) {
+      if (VIDEO_FORMATS.some((format) => clipboard.file?.endsWith(format))) {
+        setFiles([...files, clipboard.file!], true);
+      }
+    }
+  }, [clipboard]);
+
+  useEffect(() => {
+    if (selectedFinderItems) {
+      for (const item of selectedFinderItems) {
+        if (item.path && VIDEO_FORMATS.some((format) => item.path.endsWith(format))) {
+          setFiles([...files, item.path], true);
+        }
+      }
+    }
+  }, [selectedFinderItems]);
+
+  useEffect(() => {
+    for (const file of files) {
+      if (!VIDEO_FORMATS.some((format) => file.endsWith(format))) {
+        showToast({
+          title: "Invalid file",
+          message: `The file "${fileName(file)}" is not a valid video file. Supported formats are: ${VIDEO_FORMATS.join(", ")}`,
+          style: Toast.Style.Failure,
+        });
+        setFiles(files.filter((f) => f !== file));
+        return;
+      }
+    }
+  }, [files]);
+
+  const onSubmit = async () => {
+    if (files.length === 0) {
+      showToast({
+        title: "Select videos first",
+        style: Toast.Style.Failure,
+      });
+
+      return;
+    }
+    compressVideoFiles(files, compression).then((successfulFiles) => {
+      if (successfulFiles.length === 0) return;
+      successfulFiles.forEach((file) => Clipboard.copy({ file }));
+      navigation.push(<Videos files={successfulFiles} />);
```

Comment:
```suggestion
try {
        popToRoot();
        launchCommand({ name: "history", type: LaunchType.UserInitiated });
      } catch {
        /* */
      }
```
Maybe something like this instead

Reply:
what's the advantage here? I'd rather keep the current behavior because of 2 reasons:

* Passing the `successfulFiles` prop allows me to create a decorator for the new files (see screenshot below)
* If you want to go back to compress more videos, you can just press `Esc` or the *back* arrow in the UI

<img src="https://github.com/raycast/extensions/assets/4324982/a9b3c43c-84b7-4792-90c8-b9a07b5ba46a" width="33%" />

WDYT?

Reply:
That makes senes, you can maybe add a preference to let the user decide if it should popToRoot, Show compressed files(default) or just history upon completion

Reply:
I'd prefer to keep it as it is for the first version, but I'll keep it in mind for the future 🙏🏻